science/imager/files/greg.in

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh -e
+
+export GAG_GAG="${HOME}/.gag"
+export GAG_ROOT_DIR="@PREFIX@/lib/imager"
+export GAG_PATH="${GAG_ROOT_DIR}/etc"
+export GAG_EXEC_SYSTEM="@ARCH@-darwin-gfortran"
+if [ -z "\$PYTHONPATH" ]; then
+  PYTHONPATH="${GAG_ROOT_DIR}/${GAG_EXEC_SYSTEM}/python"
+else
+  PYTHONPATH="${GAG_ROOT_DIR}/${GAG_EXEC_SYSTEM}/python:${PYTHONPATH}"
+fi
+export PYTHONPATH
```

Comment:
In all four scripts, you're not actually checking whether the `PYTHONPATH` variable is empty. You're checking whether the literal string `$PYTHONPATH` is empty, which it isn't. You could fix that by not escaping the backslash:
```sh
if [ -z "$PYTHONPATH" ]; then
```
However, then you're still repeating code when your set `PYTHONPATH`. You could avoid that with something like:
```sh
[ -z "$PYTHONPATH" ] || PYTHONPATH=":$PYTHONPATH"
export PYTHONPATH="$GAG_ROOT_DIR/$GAG_EXEC_SYSTEM/python$PYTHONPATH"
```

Reply:
correction done